src/components/Details/VmDetails/VmDetails.js

```diff
@@ -0,0 +1,197 @@
+import { has, isNull } from 'lodash';
+import { VmStatus } from '../../VmStatus';
+import {
+  getCloudInitData,
+  getCpu,
+  getDescription,
+  getFlavor,
+  getMemory,
+  getNodeName,
+  getOperatingSystem,
+  getVmTemplate,
+  getWorkloadProfile,
+} from '../../../utils';
+import { CUSTOM_FLAVOR } from '../../../constants';
+
+const DASHES = '---';
+
+const cloudInitInUse = vm => !isNull(getCloudInitData(vm));
+
+const OnOffReporter = props => {
+  const statusText = props.on ? 'On' : 'Off';
+  const statusIcon = props.on ? 'pficon pficon-on' : 'pficon pficon-off';
+  const statusColor = props.on ? '#3f9c35' : '#a30000';
+
+  return (
+    <span>
+      <span style={{ color: `${statusColor}` }} className={statusIcon} /> {statusText}
+    </span>
+  );
+};
+
+OnOffReporter.propTypes = {
+  on: PropTypes.bool.isRequired,
+};
+
+const VmStatusReporter = props => {
+  const vmIsRunning = props.vm.spec.running;
+  const statusText = vmIsRunning ? 'on' : 'off';
+  const statusIcon = vmIsRunning ? 'pficon pficon-on' : 'pficon pficon-off';
+  const statusColor = vmIsRunning ? '#3f9c35' : '#a30000';
+
+  return (
+    <span style={{ color: `${statusColor}` }}>
+      <span className={statusIcon} /> {props.vm.metadata.name} is {statusText}
+    </span>
+  );
+};
+
+VmStatusReporter.propTypes = {
+  vm: PropTypes.object.isRequired,
+};
+
+export const getVmIpAddresses = vmi => {
+  let ipAddresses = [];
+  if (has(vmi, 'status.interfaces')) {
+    ipAddresses = vmi.status.interfaces.map(iface => iface.ipAddress);
+  }
+
+  return ipAddresses;
+};
+
+const Flavor = props => {
+  const { vm } = props;
+  const flavor = getFlavor(vm);
+  if (!flavor) {
+    return DASHES;
+  }
+  const isCustomFlavor = flavor === CUSTOM_FLAVOR;
+
+  let resourceElement = null;
+  if (isCustomFlavor) {
+    const cpu = getCpu(vm);
+    const memory = getMemory(vm);
+    let resourceStr = '';
+    resourceStr += cpu ? `${cpu} CPU` : '';
+    resourceStr += memory ? `, ${memory} Memory` : '';
+    resourceElement = <div>{resourceStr}</div>;
+  }
+
+  return (
+    <Fragment>
+      <div>{flavor}</div>
+      {resourceElement}
+    </Fragment>
+  );
+};
+
+Flavor.propTypes = {
+  vm: PropTypes.object.isRequired,
+};
+
+export const VmDetails = props => {
+  const { launcherPod, importerPod, migration, NodeLink, vm, vmi, PodResourceLink, NamespaceResourceLink } = props;
+  const nodeName = getNodeName(launcherPod);
+  const description = getDescription(vm);
+  const ipAddresses = getVmIpAddresses(vmi);
+
+  return (
+    <Fragment>
+      <div className="co-m-pane__body">
+        <h1 className="co-m-pane__heading">Overview</h1>
+        <div>
+          <div className="row">
+            {/* Description column */}
+            <div className="col-sm-4">
+              <div>
+                <VmStatusReporter vm={vm} />
+              </div>
+              {!description || <div style={{ marginTop: '10px', marginRight: '40px' }}>{description}</div>}
```

Comment:
it would be nice to show something like `VM has no description` in case there isnt any

Reply:
I had it showing dashes at one point, but Tomas pointed out that it looked weird (it didn't have a "Description" label at the time). I added the text you mention, which looks much better since there is a label now.